app/controllers/posts_controller.rb

```diff
@@ -175,6 +176,7 @@ def delete_history
   end
 
   def stats
+    fresh_when(etag: @post, last_modified: @post.tagged_at, public: false)
```

Comment:
We should use just `@post.updated_at` here, I think – the main post could be updated, changing its word count, without updating `tagged_at`, but changing the contents of this page. The `object:` param should handle both etag and last_modified correctly:

```suggestion
    fresh_when(object: @post, public: false)
```